internal/auth/credentials.go

```
@@ -208,22 +209,28 @@ func GetNewCredentialsWithMetadata(m map[string]interface{}, tokenSecret string)
 	// Generate access key.
 	keyBytes, err := readBytes(accessKeyMaxLen)
 	if err != nil {
-		return cred, err
+		return "", "", err
 	}
 	for i := 0; i < accessKeyMaxLen; i++ {
 		keyBytes[i] = alphaNumericTable[keyBytes[i]%alphaNumericTableLen]
 	}
-	accessKey := string(keyBytes)
+	accessKey = string(keyBytes)
 
 	// Generate secret key.
 	keyBytes, err = readBytes(secretKeyMaxLen)
 	if err != nil {
-		return cred, err
+		return "", "", err
 	}
 
-	secretKey := strings.Replace(string([]byte(base64.StdEncoding.EncodeToString(keyBytes))[:secretKeyMaxLen]),
+	secretKey = strings.Replace(string([]byte(base64.StdEncoding.EncodeToString(keyBytes))[:secretKeyMaxLen]),
 		"/", "+", -1)
 
+	return accessKey, secretKey, nil
+}
+
+// GetNewCredentialsWithMetadata generates and returns new credential with expiry.
+func GetNewCredentialsWithMetadata(m map[string]interface{}, tokenSecret string) (cred Credentials, err error) {
+	accessKey, secretKey, err := GenerateCredentials()
```

You need to return an error here if err != nil

---

Good catch! Fixed.